tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java

```diff
@@ -64,13 +72,44 @@ public TsFileSequenceReader(String file) throws IOException {
    * @param loadMetadataSize -load meta data size
    */
   public TsFileSequenceReader(String file, boolean loadMetadataSize) throws IOException {
+    this(file, loadMetadataSize, false);
+  }
+
+  /**
+   * construct function for TsFileSequenceReader.
+   *
+   * @param file -given file name
+   * @param loadMetadataSize -load meta data size
+   * @param autoRepair if true it tries to automatically repair a damaged file (by changing it!)
+   *                   before reading
+   */
+  public TsFileSequenceReader(String file, boolean loadMetadataSize, boolean autoRepair) throws IOException {
     this.file = file;
-    tsFileInput = new DefaultTsFileInput(Paths.get(file));
+    final Path path = Paths.get(file);
+    tsFileInput = new DefaultTsFileInput(path);
+    if (autoRepair) {
+      checkAndRepair(file, path);
+    }
     if (loadMetadataSize) {
       loadMetadataSize();
     }
   }
 
+  /**
+   * Checks if the file is incomplete, and if so, tries to repair it.
+   */
+  private void checkAndRepair(String file, Path path) throws IOException {
+    // Check if file is damaged
+    if (!TSFileConfig.MAGIC_STRING.equals(readTailMagic())) {
```

Comment:
a minor bug:
if the file only has a headMagicString (i.e., TsFilev0.8.0), then checkAndRepair() will consider it a complete TsFile...
So, we need to confirm that the file.length > MAGIC_STRING.length..

Reply:
Hey @jixuan1989 you are, indeed, right.
I decided to move this check directly to the readTailMagic() function as this belongs there from my perspective.
Otherwise I tell it to readTailMagic() and it reads header magic which I then check due to implementation knowledge.
For me its more intuitive it readTailMagic throws an Exception if it sees that it cannot do so.
What do you think?

Reply:
agree!